roles/tas_single_node/tasks/certificates.yml

```diff
@@ -93,6 +88,16 @@
           -out '{{ tas_single_node_remote_fulcio_private_key }}'
           -passout 'pass:{{ tas_single_node_fulcio_ca_passphrase }}'
         creates: "{{ tas_single_node_remote_fulcio_private_key }}"
+      when: >
+        tas_single_node_fulcio_private_key == ""
+        and (certs_dir_files.files | selectattr('path', 'equalto', tas_single_node_remote_fulcio_private_key) | list | length) == 0
+
+    - name: Load user-provided Fulcio private key
+      ansible.builtin.copy:
+        src: "{{ tas_single_node_fulcio_private_key }}"
+        dest: "{{ tas_single_node_remote_fulcio_private_key }}"
+        mode: '0600'
+      when: tas_single_node_fulcio_private_key != ""
```

Comment:
For the two private key tasks, I think we should:
* Use a different path for the key file for each task. For example, if the user first provided a key themselves but later wanted to switch to a role-generated key, we would just happily keep using their provided old key. Using different paths for the file should solve this.
* For the user-provided key, I think if it's not provided we should explicitly ensure that the file is absent (we don't want users credentials laying around on the node unnecessarily).
* Based on what the user configured, we can then copy either the role generated key or the user key to `tas_single_node_remote_fulcio_private_key` as the final destination (so we won't have to change any tasks that depend on that file location).